src/node/rpc/node_call_types.h

```diff
@@ -105,25 +104,31 @@ namespace ccf
 
         LedgerSecretsMap ledger_secrets;
         NetworkIdentity identity;
-        ServiceStatus service_status;
-
-        bool operator==(const NetworkInfo& other) const
-        {
-          return public_only == other.public_only &&
-            last_recovered_signed_idx == other.last_recovered_signed_idx &&
-            consensus_type == other.consensus_type &&
-            ledger_secrets == other.ledger_secrets &&
-            service_status == other.service_status &&
-            identity == other.identity;
-        }
-
-        bool operator!=(const NetworkInfo& other) const
-        {
-          return !(*this == other);
-        }
+        std::optional<ServiceStatus> service_status = std::nullopt;
+
+        NetworkInfo() {}
+
+        NetworkInfo(
+          bool public_only,
+          kv::Version last_recovered_signed_idx,
+          ConsensusType consensus_type,
+          const LedgerSecretsMap& ledger_secrets,
+          const NetworkIdentity& identity,
+          ServiceStatus service_status) :
+          public_only(public_only),
+          last_recovered_signed_idx(last_recovered_signed_idx),
+          consensus_type(consensus_type),
+          ledger_secrets(ledger_secrets),
+          identity(identity),
+          service_status(service_status)
+        {}
+
+        bool operator==(const NetworkInfo& other) const = default;
```

Comment:
This won't work with C++17

Reply:
Is the plan not to break compatibility with C++17 just yet?

Reply:
As discussed, we won't break C++!7 compatibility just yet, until we know that existing CCF apps can upgrade to C++20 safely.